src/pybind/mgr/cephadm/cert_mgr.py

```diff
@@ -3,103 +3,126 @@
 import copy
 
 from cephadm.ssl_cert_utils import SSLCerts, SSLConfigException
-from typing import TYPE_CHECKING, Tuple, Union, List, Optional
+from mgr_util import verify_tls, ServerConfigException
+from cephadm.ssl_cert_utils import get_certificate_info, get_private_key_info
 from cephadm.tlsobject_types import Cert, PrivKey
-from cephadm.tlsobject_store import TLSObjectStore, TLSObjectScope
+from cephadm.tlsobject_store import TLSObjectStore, TLSObjectScope, TLSObjectException
 
 if TYPE_CHECKING:
     from cephadm.module import CephadmOrchestrator
 
 logger = logging.getLogger(__name__)
 
 
+class CertInfo:
+    """
+      - is_valid: True if the certificate is valid.
+      - is_close_to_expiration: True if the certificate is close to expiration.
+      - days_to_expiration: Number of days until expiration.
+      - error_info: Details of any exception encountered during validation.
+    """
+    def __init__(self, cert_name: str,
+                 target: Optional[str],
+                 user_made: bool = False,
+                 is_valid: bool = False,
+                 is_close_to_expiration: bool = False,
+                 days_to_expiration: int = 0,
+                 error_info: str = ''):
+        self.user_made = user_made
+        self.cert_name = cert_name
+        self.target = target or ''
+        self.is_valid = is_valid
+        self.is_close_to_expiration = is_close_to_expiration
+        self.days_to_expiration = days_to_expiration
+        self.error_info = error_info
+
+    def __str__(self) -> str:
+        return f'{self.cert_name} ({self.target})' if self.target else f'{self.cert_name}'
+
+    def is_operationally_valid(self) -> bool:
+        return self.is_valid and not self.is_close_to_expiration
+
+    def get_status_description(self) -> str:
+        cert_source = 'user-made' if self.user_made else 'self-signed'
```

Comment:
NIT, would `cephadm-generated` be a better thing to call this rather than self-signed? Technically a user made cert could be self signed as well I think.

Reply:
Good point, I think your suggestion makes a lot of sense. As you said the term `self-signed` can be misleading. I'll change it.